pkgs/top-level/splice.nix

```diff
@@ -17,6 +17,20 @@
 lib: pkgs: actuallySplice:
 
 let
+  inherit (lib)
+    attrByPath
+    callPackageWith
+    callPackagesWith
+    genAttrs
+    isAttrs
+    isDerivation
+    listToAttrs
+    makeScopeWithSplicing
+    makeScopeWithSplicing'
+    optional
+    optionalAttrs
+    splitString
+    ;
```

Comment:
You should be able to hoist this between the `lib:` and `pkgs:` so it's shared over a wider context, too.